### PR TITLE
Editor - Integrate visual editor 0.5

### PR DESCRIPTION
**Pull request by aforcier**
Integrates the [0.5](https://github.com/wordpress-mobile/WordPress-Editor-Android/milestones/0.5) milestone of the visual editor, excluding galleries (those will be added in a separate PR once this one is merged).

Changes:
* Added support for creating a new post by sharing content from other apps and from the Media Gallery ([PR](https://github.com/wordpress-mobile/WordPress-Editor-Android/pull/262))
* Navigation fixes for the image settings dialog ([PR](https://github.com/wordpress-mobile/WordPress-Editor-Android/pull/267) for details and screenshots):
  * Device back button correctly returns to the editor instead of returning to the post list
  * A confirmation dialog is shown if attempting to leave the dialog with unsaved changes
  * A toast is shown confirming that changes were saved when the Save button is pressed

* Resolves breaking changes introduced by merging `develop` into the `feature/visual-editor` branch (PR #3449's changes to media events in particular)

Note: Featured image support is also part of the milestone, but it was already added in #3425

### CODE REVIEW

File: WordPress/src/main/java/org/wordpress/android/ui/media/services/MediaUploadService.java [hunk 1]
```diff
@@ -170,18 +171,20 @@ private void uploadMediaFile(Cursor cursor) {
...
+                EventBus.getDefault().post(new MediaEvents.MediaUploadSucceeded(blogIdStr, mediaId,
+                        remoteId, remoteUrl));
+                fetchMediaFile(remoteId);
             }
 
             @Override
             public void onFailure(ApiHelper.ErrorType errorType, String errorMessage, Throwable throwable) {
                 WordPress.wpDB.updateMediaUploadState(blogIdStr, mediaId, MediaUploadState.FAILED);
                 mUploadInProgress = false;
+                mCurrentUploadMediaId = "";
```
Comment (maxme) [inline WordPress/src/main/java/org/wordpress/android/ui/media/services/MediaUploadService.java]: good catch

**Comment by maxme** [comment]
We must include that fix https://github.com/wordpress-mobile/WordPress-Android/pull/3455  in `WordPressMediaUtils.loadNetworkImage`

**Comment by aforcier** [comment]
> We must include that fix #3455 in `WordPressMediaUtils.loadNetworkImage`

Well spotted, fixed in 887bf66.

**Comment by maxme** [comment]
:shipit:

**Pull request merged**